lib/client/terminal/terminal_windows.go

```diff
@@ -0,0 +1,279 @@
+//go:build windows
+// +build windows
+
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package terminal
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+	"syscall"
+
+	"github.com/Azure/go-ansiterm/winterm"
+	"github.com/gravitational/teleport/lib/client/tncon"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+	"github.com/moby/term"
+)
+
+// initTerminal configures the terminal for raw, VT compatible output and
+// optionally input. The returned function should be called before program
+// exit to ensure the terminal is reset, otherwise it will be left in a broken
+// state.
+func initTerminal(input bool) (func(), error) {
+	stdoutFd := int(syscall.Stdout)
+	stdinFd := int(syscall.Stdin)
+
+	oldOutMode, err := winterm.GetConsoleMode(uintptr(stdoutFd))
+	if err != nil {
+		return func() {}, fmt.Errorf("failed to retrieve stdout mode: %w", err)
```

Comment:
I don't have a strong preference, but you can return nil instead of a nop func.

Reply:
nil returns seem to generate nil pointer dereferences (https://play.golang.org/p/8IFuyejGiVt) so the nop func seems like a nicer thing to hand back to callers.

Reply:
Well yes you can't call the nil func, but I assumed the caller would do an error check first and only defer the cleanup if err == nil. Either way is fine though.